### DISCUSSION TITLE
Discussion items for meeting

### DISCUSSION THREAD

**Discussion by OOHehir** [LIFsCode/ELOC-3.0#81] [General]
@EDsteve @LIFsCode Two ideas I though I'd bring up at the meeting:
1. Reorganization of file & directories. 
I was considering moving some of the EdgeImpulse code from the /src directory into the /lib directory. It'll allow the code to be unit tested. @LIFsCode Would you have any thoughts about doing similar with the the other components? Correcting the various #include's will probably be a bit of work so makes sense to do it in a coordinated fashion rather than piecemeal.
2. OTA Updating
I'll hopefully have some of the system testing setup in the next few weeks, but a significant element of the CI/CD setup would to enable an autonomous firmware updating process. This seems to be implemented from the SD card but that doesn't allow autonomous updating. Has OTA updating being considered before? If power usage is the concern for WiFi it could just be enabled on boot-up for a few seconds, or perhaps when a file is present on the SD card. The device would search for a predetermined WiFi SSID & connect & attempt to download the latest image (from GitHub perhaps). Bluetooth also seems to be an option. Increase in code size may be an issue (I haven't examined the impact) but enabling such updating would seem to be a useful feature.

**Comment by EDsteve** [comment]
@OOHehir 
OTA updates from the phone via BT is on our feature list for the future. If you integrate it for testing. Maybe that could be done at the same time for the app? No high priority. Just thinking out loud :) 

@LIFsCode @OOHehir 
I just got some not so good news from the jungle. From 8 ELOCs only 4 ELOCs are still recording after 2 days. I never had so many devices stopping. So i hope the problem is caused by something else. Will hopefully have more details later.

**Reply by OOHehir** [reply]
@EDsteve That's disappointing to hear.

Thanks for the information on the BT. I'll look into it.
